EIPS/eip-4950.md

```diff
@@ -0,0 +1,86 @@
+---
+eip: 4950
+title: Entangled Tokens
+description: ERC-721 extension with two tokens minted that are tied together
+author: Victor Munoz (@victormunoz)
+discussions-to: https://ethereum-magicians.org/t/entangled-tokens/8702
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-03-28
+requires: 721
+---
+
+## Abstract
+This EIP standardizes an interface for creating entangled pairs of tokens. An entangled pair of tokens give his owners the right to operate with this common wallet.
+
+## Motivation
+The motivation is to provide an easy way to share a wallet through NFTs. So that the act of buying an NFT (in a marketplace) gives the buyer the privilege to have access to a given wallet. This wallet could have budget in many tokens, or even be the propietary of other NFTs.
+
+An use case is to keep contact between an artist and an buyer of its NFTs. If an artist T has created a digital piece of art P with an NFT, then T creates 2 entangled tokens A and B so that he keeps A and transfer B to P. By construction of entangled tokens, only one transfer is possible for them, thus the artist proofs he’s been the creator of P by sending a transaction to A that is visible from B. Otherwise, the owner of P might check the authenticity of the artist by sending a transaction to B so that the artist might proof by showing the outcome out of A.

```

Comment:
Can't you check that `T` is the creator of an NFT by looking at the transaction history, or am I missing something?

Reply:
The point is that if the NFT buyer sees that something changed (for example the balance) in the shared wallet (shared with the artist T), then it means that the artist has been the author of that change. Because only the artist (and himself) can make changes to that shared wallet.
And viceversa (if the buyer sends something to the shared wallet, for example a donation, the the artist can withdraw it).

They are "connected" through the entangled tokens. They don't need to check in any blockchain explorer for the transaction history. It is implicit. And in the future you could even make this check programatically (from other contracts).